src/containers/main/ReferContainer/index.tsx

```diff
@@ -31,15 +31,26 @@ export const ReferContainer: React.FC<IProps> = () => {
 
 	useEffect(() => {
 		let isMounted = true;
+		const allCookies = document.cookie ?? '';
+
+		if (allCookies && allCookies.length > 1) {
+			setReferredBy(
+				allCookies
+					?.split('; ')
+					?.find((row) => row.startsWith('referralId='))
+					?.split('=')[1],
+			);
+		}
+
 		if (address && contract) {
 			(async () => {
 				const contractMethods = await contract.methods;
 				const affiliate = await contractMethods.customerToAffiliate(address.toLowerCase()).call();
 				if (affiliate !== '0') {
-					const referral = await contractMethods.affiliates(affiliate).call();
-					if (isMounted) {
-						setReferredBy(referral.name);
-					}
+					// const referral = await contractMethods.affiliates(affiliate).call();
+					// if (isMounted) {
+					// 	setReferredBy(referral.name);
+					// }
```

Comment:
if this chunk of code isn't needed, remove it

Reply:
I don't think you can get rid of this like this, I visited the page with an account that was already referred by another affiliate and it shows me that I'm organic even though Im not, its because you just commented this out. If someone was already referred, youre code will just show them as organic unless the cookie is set, if the cookie is set, it will show the referral id of the wrong affiliate. 

TLDR, you don't account for the senario someone was already referred, commenting this out isn't the right thing to do to resolve this issue